react/features/welcome/components/WelcomePage.native.js

```diff
@@ -65,49 +91,88 @@ class WelcomePage extends AbstractWelcomePage {
+                        <View style = { styles.audioVideoSwitchContainer }>
+                            <Text style = { PlatformElements.headerText } >
+                                Audio
+                            </Text>
+                            <Switch
+                                onTintColor = { SWITCH_UNDER_COLOR }
+                                onValueChange = { this._onStartAudioOnlyChange }
+                                style = { styles.audioVideoSwitch }
+                                thumbTintColor = { SWITCH_THUMB_COLOR }
+                                value = { !_profile.startAudioOnly } />
+                            <Text style = { PlatformElements.headerText } >
+                                Video
+                            </Text>
+                        </View>
+                    </Header>
+                    <SafeAreaView style = { styles.roomContainer } >
+                        <TextInput
+                            accessibilityLabel = { 'Input room name.' }
+                            autoCapitalize = 'none'
+                            autoComplete = { false }
+                            autoCorrect = { false }
+                            autoFocus = { false }
+                            onChangeText = { this._onRoomChange }
+                            onSubmitEditing = { this._onJoin }
+                            placeholder = { t('welcomepage.roomname') }
+                            placeholderTextColor = { PLACEHOLDER_TEXT_COLOR }
+                            returnKeyType = { 'go' }
+                            style = { styles.textInput }
+                            underlineColorAndroid = 'transparent'
+                            value = { this.state.room } />
                         {
                             this._renderJoinButton()
                         }
-                    </View>
-                    <RecentList />
+                        <RecentList />
+                    </SafeAreaView>
+                    <AppSettings />
                 </View>
-                <AppSettings />
-                {
-                    this._renderLegalese()
-                }
+                <WelcomePageSideBar />
             </LocalVideoTrackUnderlay>
         );
     }
 
+    /**
+     * Toggles the side bar.
+     *
+     * @private
+     * @returns {void}
+     */
+    _onShowSideBar() {
+        this.props.dispatch(setSideBarVisibility(true));
+    }
+
+    /**
+     * Handles the audio-video switch changes.
+     *
+     * @private
+     * @param {boolean} startWithVideo - The negaty of the new value
+     * of startAudioOnly.
+     * @returns {void}
+     */
+    _onStartAudioOnlyChange(startWithVideo) {
+        const { dispatch } = this.props;
+
+        dispatch(updateProfile({
+            ...this.props._profile,
+            startAudioOnly: !startWithVideo
+        }));
+
+        dispatch(setAudioOnly(!startWithVideo));
```

Comment:
I think this belongs in the profile middleware. This way, if / when we add the ability to change the profile mid-conference, it will Just Work.